app/src/main/java/net/aiscope/gdd_app/repository/SampleRepositorySharedPreference.kt

```diff
@@ -47,4 +49,15 @@ class SampleRepositorySharedPreference @Inject constructor(
             gson.fromJson<SampleDto>(it).toDomain()
         }.toList()
     }
+
+    override suspend fun last(): Sample? {
+        val allStores = all()
+        val facility = healthFacilityRepository.load()
+        return allStores
+            .filter{s -> s.healthFacility == facility.id
+                        && s.microscopist == facility.microscopist
+                        && s.createdOn != null
+                        && s.status != Status.Incomplete}
+            .sortedWith(compareBy {it.createdOn}).lastOrNull()
```

Comment:
Could be written shorter
```suggestion
            .sortedBy { it.createdOn }.lastOrNull()
```

Reply:
Thanks for this suggestion! I'm making this change.